File: e-commerce(final project)/inventory-service/src/main/java/com/fiba/inventoryservice/data/entities/Product.java
package com.fiba.inventoryservice.data.entities;



import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter @Setter @NoArgsConstructor
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;
    private String productName;
    private double salesPrice;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="category_id")
    private Category category;

    public Product(String productName, double salesPrice) {
        this.productName = productName;
        this.salesPrice = salesPrice;
    }

}

File: e-commerce(final project)/AngularClient/fiba-ui/src/app/models/Product.ts

export class Product {
    productId:number;
    productName:string;
    salesPrice: number;
    categoryName:string;
    constructor(productId:number,productName:string,salesPrice: number,categoryName:string){
      this.productId = productId;
      this.productName = productName;
      this.salesPrice =salesPrice;
      this.categoryName = categoryName; 
    }
}